Core/Core/Offline/OfflineService.swift

```diff
@@ -47,3 +47,22 @@ public final class OfflineServiceLive: OfflineService {
         !availabilityService.status.isConnected
     }
 }
+
+class OfflineServiceModel: ObservableObject {
+    @Published var isOffline: Bool = false
+
+    private let networkAvailabilityService = NetworkAvailabilityServiceLive()
+    private var subscriptions = Set<AnyCancellable>()
+
+    init(offlineService: OfflineServiceLive) {
+        unowned let unownedSelf = self
+        networkAvailabilityService.startMonitoring()
+        networkAvailabilityService
+            .startObservingStatus()
+            .receive(on: DispatchQueue.main)
+            .sink {
+                unownedSelf.isOffline = $0 == .disconnected && offlineService.isOfflineModeEnabled()
+            }
+            .store(in: &subscriptions)
+    }
+}
```

Comment:
This seems like an object that should be an interactor in the current architecture. We could probably just use one interactor for this business logic. 
We should inject `NetworkAvailabilityServiceLive` so it's testable.

Reply:
I'd rename `OfflineService` to be an interactor, add reactive functions  and make every view's view model depend on this interactor.